### PR TITLE
Cody completions: Fix interop between spaces and tabs

### PR DESCRIPTION
**Pull request by philipp-spiess**
This PR fixes two issues related to spaces/tabs interop:

1. **Odd indentation** (= when Cody inserts a sole space at the beginning of an indented code line) is inserted after a current eventual tab indentation (so it looks like this: `\t\t something`). To account for that, the regex now captures preceeding tabs and keeps them in tact.
2. **Tab to space converstion**. Today Cody started to respond with tab indented text (for no reason?). I want to leave it to the IDE to conver this but we have to account for this in the indentation function (where we cound the indentaiton for a line).
 
## Test plan

- Manually verified
- I really need to add the test suite for this. Will come very very soon I promise 🤞

### CODE REVIEW

**Comment by abeatrix** [comment]
I just tried to run it locally. Do you know why Cody is suggesting code with `//` prefix every other line?
image

**Comment by abeatrix** [comment]
And if I remove the `//` the indentation looks off, not sure if this is related to the issue that this PR is trying to fix:
image

**Comment by philipp-spiess** [comment]
@abeatrix can you include the logs please? want to rule out it's the backends response.

**Comment by abeatrix** [comment]
> @abeatrix can you include the logs please? want to rule out it's the backends response.
> 
> Also I’m a bit confused from the screenshot on what is a code comment and what is a completion 😅

yea so I added line 18 manually, and then go to the next line,
- line 19 : cody produces the code without issue so I hit tab then enter
- line 20: code suggested new code but the code started with `//`, again hit tab then enter
- line 21: new code without `//`
- line 22: with `//` again

Will send the log in a few

**Comment by philipp-spiess** [comment]
@abeatrix and me have chatted about this issue in https://sourcegraph.slack.com/archives/C052G9Y5Y8H/p1685117047371809?thread_ts=1685114485.854269&cid=C052G9Y5Y8H.

**Comment by philipp-spiess** [comment]
@abeatrix Do you happen to have logs for the indentation issue was well now that your setup works? The last screenshot you sent me was unrelated to indentation issues

File: client/cody/src/completions/provider.ts [hunk 1]
```diff
@@ -469,9 +469,21 @@ export function sliceUntilFirstNLinesOfSuffixMatch(suggestion: string, suffix: s
...
 
 /**
- * Counts space or tabs in the beginning of a line
+ * Counts space or tabs in the beginning of a line.
+ *
+ * Since Cody can sometimes respond in a mix of tab and spaces, this function
+ * normalizes the whitespace first using the currently enabled tabSize option.
  */
 function indentation(line: string): number {
+    const tabSize: number = vscode.window.activeTextEditor
```
Comment (valerybugakov) [inline client/cody/src/completions/provider.ts:478]:
```suggestion
    const tabSize = vscode.window.activeTextEditor
```

**Review by valerybugakov — APPROVED** [review]

**Review by valerybugakov — APPROVED** [review]

**Pull request merged**